engine/src/main/scala/cromwell/engine/io/nio/NioFlow.scala

```diff
@@ -177,31 +173,7 @@ class NioFlow(parallelism: Int,
     }
 
   private def hash(hash: IoHashCommand): IO[String] =
-    // If there is no hash accessible from the file storage system,
-    // we'll read the file and generate the hash ourselves.
-    getStoredHash(hash.file)
-      .flatMap {
-        case Some(storedHash) => IO.pure(storedHash)
-        case None => generateMd5FileHashForPath(hash.file)
-      }
-      .map(_.hash)
-
-  private def getStoredHash(file: Path): IO[Option[FileHash]] =
-    file match {
-      case gcsPath: GcsPath => getFileHashForGcsPath(gcsPath).map(Option(_))
-      case blobPath: BlobPath => getFileHashForBlobPath(blobPath)
-      case drsPath: DrsPath =>
-        IO {
-          // We assume all DRS files have a stored hash; this will throw
-          // if the file does not.
-          drsPath.getFileHash
-        }.map(Option(_))
-      case s3Path: S3Path =>
-        IO {
-          Option(FileHash(HashType.S3Etag, s3Path.eTag))
-        }
-      case _ => IO.pure(None)
-    }
+    NioHashing.hash(hash.file)
```

Comment:
The `NioFlow` class is supposed to handle all 10 file methods but it was getting to be mostly hashing, hence the split-out.

It also wasn't really clear what hashing methods you were actually supposed to call, and which ones existed solely for use by other hashing-related methods. Now the interface is clear, you can either `hash` or `getStoredHash`.